ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp

```diff
@@ -123,10 +123,8 @@ auto preprocess_inputs(
 
     return [](const auto &input_tensor_a, const auto &input_tensor_b) {
         if constexpr (detail::is_associative(binary_op_type)) {
-            const auto input_shape_a = input_tensor_a.get_shape();
-            const auto input_shape_b = input_tensor_b.get_shape();
             // Swap tensors if input_tensor_a needs to be broadcasted to input_tensor_b
-            if (tt::tt_metal::compute_volume(input_shape_a) < tt::tt_metal::compute_volume(input_shape_b)) {
+            if (input_tensor_a.get_logical_volume() < input_tensor_b.get_logical_volume()) {
```

Comment:
My understanding that in current change:
* volume -> returns physical volume
* get_logical_volume -> returns logical volume

I think in future, we should swap that:
* volume -> should return logical volume
* get_physical_volume -> returns physical volume

Again, feel free to ignore this comment. I've seen that this will be done in next PRs.

Reply:
* we should rename volume to get_physical_volume
* we can rename get_logical_volume to get_volume or can leave as is or make it a method in SimpleShape

Reply:
Can do in a separate PR after we get through the main body of changes. Thank you!